tests/project/_common/test_movement.py

```diff
@@ -0,0 +1,437 @@
+            )
+        # test trying to write an existing unified dataset name
+        with pytest.raises(ValueError):
+            categorization.import_artifacts(
+                project_artifact_path=artifact_path,
+                tamr_client=client,
+                new_project_name=new_project_name,
+                new_unified_dataset_name="minimal_categorization_unified_dataset",
+            )
+        if new_project_name in [p.name for p in client.projects.stream()]:
+            raise RuntimeError(f"{new_project_name} is being unintentionally created during test.")
+    else:
+        raise AssertionError(f"{new_project_name} already exists in test instance.")
+
+
+@mock_api()
+def test_import_existing_errors():
+    client = utils.client.create(**CONFIG["toolbox_test_instance"])
+    project = client.projects.by_resource_id(CONFIG["projects"]["minimal_categorization"])
+
+    # export a project
+    op = categorization.export_artifacts(
+        project=project,
+        artifact_directory_path="/home/ubuntu/tamr/projectExports",
+        exclude_artifacts=None,
+        asynchronous=False,
+    )
+
+    assert op.succeeded()
+
+    # get artifact path
+    description = op.description
+    artifact_path = str(description).split(" ")[-1]
+    print(artifact_path)
+
+    assert artifact_path[0] == "/"
+    assert artifact_path[-4:] == ".zip"
+
+    # get existing project
+    existing_project = client.projects.by_name("minimal_incomplete_categorization")
+
+    # test trying to set new_project name on existing project
+    with pytest.raises(KeyError):
+        categorization.import_artifacts(
+            tamr_client=existing_project.client,
+            project_artifact_path=artifact_path,
+            target_project=existing_project,
+            new_project_name="new_project_name",
+            overwrite_existing=True,
+        )
+
+    # test trying to set new_unified_dataset_name on existing project
+    with pytest.raises(KeyError):
+        categorization.import_artifacts(
+            tamr_client=existing_project.client,
+            project_artifact_path=artifact_path,
+            target_project=existing_project,
+            new_unified_dataset_name="new_ud_name",
+            overwrite_existing=True,
+        )
+
+    # test overwrite existing
+    with pytest.raises(KeyError):
+        categorization.import_artifacts(
+            tamr_client=existing_project.client,
+            project_artifact_path="incorrect/artifact/path",
+            target_project=existing_project,
+            overwrite_existing=False,
+        )
+
+    # test incorrect artifact path
+    with pytest.raises(ValueError):
+        categorization.import_artifacts(
+            tamr_client=existing_project.client,
+            project_artifact_path="incorrect/artifact/path",
+            target_project=existing_project,
+            overwrite_existing=True,
+        )
+
+    # fail if not present
+    with pytest.raises(ValueError):
```

Comment:
Shouldn't this always fail if not present? What is the current behavior?

Reply:
My understanding is that, by default, artifacts can be imported into a project even if those artifacts do not exist in the project. When the failIfNotPresent flag does is it makes the import fail if an artifact exists in the zip which does not exist in the target project.